src/MamoStakingStrategy.sol

```diff
@@ -0,0 +1,285 @@
+     */
+    function initialize(InitParams calldata params) external initializer {
+        require(params.mamoStrategyRegistry != address(0), "Invalid mamoStrategyRegistry address");
+        require(params.stakingRegistry != address(0), "Invalid stakingRegistry address");
+        require(params.multiRewards != address(0), "Invalid multiRewards address");
+        require(params.mamoToken != address(0), "Invalid mamoToken address");
+        require(params.strategyTypeId != 0, "Strategy type id not set");
+        require(params.owner != address(0), "Invalid owner address");
+
+        __BaseStrategy_init(params.mamoStrategyRegistry, params.strategyTypeId, params.owner);
+
+        stakingRegistry = MamoStakingRegistry(params.stakingRegistry);
+        multiRewards = IMultiRewards(params.multiRewards);
+        mamoToken = IERC20(params.mamoToken);
+    }
+
+    /**
+     * @notice Set slippage tolerance for this strategy
+     * @param slippageInBps The slippage tolerance in basis points (e.g., 100 = 1%)
+     */
+    function setAccountSlippage(uint256 slippageInBps) external onlyOwner {
+        require(slippageInBps <= stakingRegistry.MAX_SLIPPAGE_IN_BPS(), "Slippage too high");
+
+        emit AccountSlippageUpdated(accountSlippageInBps, slippageInBps);
+        accountSlippageInBps = slippageInBps;
+    }
+
+    /**
+     * @notice Get the slippage tolerance for this strategy
+     * @return The slippage tolerance in basis points (falls back to global if not set)
+     */
+    function getAccountSlippage() public view returns (uint256) {
+        return accountSlippageInBps > 0 ? accountSlippageInBps : stakingRegistry.defaultSlippageInBps();
+    }
+
+    /**
+     * @notice Deposit MAMO tokens into MultiRewards (permissionless)
+     * @param amount The amount of MAMO to deposit
+     */
+    function deposit(uint256 amount) external {
+        require(amount > 0, "Amount must be greater than 0");
+
+        // Transfer MAMO from depositor to this contract
+        mamoToken.safeTransferFrom(msg.sender, address(this), amount);
+
+        _stakeMamo(amount);
+
+        emit Deposited(msg.sender, amount);
+    }
+
+    /**
+     * @notice Withdraw MAMO tokens from MultiRewards
+     * @param amount The amount of MAMO to withdraw
+     */
+    function withdraw(uint256 amount) external onlyOwner {
+        require(amount > 0, "Amount must be greater than 0");
+
+        // Withdraw from MultiRewards
+        multiRewards.withdraw(amount);
+
+        // Transfer withdrawn MAMO to owner
+        mamoToken.safeTransfer(msg.sender, amount);
+
+        emit Withdrawn(amount);
+    }
+
+    /**
+     * @notice Withdraw all staked MAMO tokens from MultiRewards
+     */
+    function withdrawAll() external onlyOwner {
+        uint256 stakedBalance = multiRewards.balanceOf(address(this));
+        require(stakedBalance > 0, "No tokens to withdraw");
+
+        // Withdraw all from MultiRewards
+        multiRewards.withdraw(stakedBalance);
```

Comment:
what happens to the users pending rewards in multirewards

Reply:
We should probably call `exit` instead of `withdraw` exit withdraws all and gets the reward

Reply:
nice catch